pkg/middleware/validation.go

```diff
@@ -60,7 +60,17 @@ func validate(hook, mod, ver string) (bool, error) {
 		return false, errors.E(op, err)
 	}
 
-	resp, err := http.Post(hook, "application/json", bytes.NewBuffer(jsonVal))
+	req, err := http.NewRequest(http.MethodPost, hook, bytes.NewBuffer(jsonVal))
+	if err != nil {
+		return false, errors.E(op, err)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	if auth != "" {
+		req.Header.Add("Authorization", auth)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Could you use a custom http client here instead so that we can pass a timeout?

Reply:
as another, more high level, option: you can do `req = req.WIthContext(ctx)` and have a timeout on the context. 

Either one is fine imo

Reply:
Would it be ok with folks if we merge this and add the timeout (fwiw I like the `req.WithContext` approach better) in a follow-up?